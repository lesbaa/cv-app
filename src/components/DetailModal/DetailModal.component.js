import React from 'react'
import styles from './DetailModal.styles'

const DetailModal = ({
  id,
  modalTitle,
  modalContent,
  modalIsVisible,
  originX,
  originY,
  hideSkillDetailModal,
}) => {
  const detailModalClassNames = [
    'DetailModal',
    modalIsVisible && 'is-open',
  ]
    .filter(Boolean)
    .join(' ')

  return (
    <aside
      className={detailModalClassNames}
      onClick={hideSkillDetailModal}
      style={{
        transform: modalIsVisible && `translate(${originX}px,${originY}px)`,
      }}
    >
      <style jsx>{styles}</style>
    </aside>
  )
}

export default DetailModal
// TODO proptypes