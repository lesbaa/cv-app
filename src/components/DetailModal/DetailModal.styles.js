import css from 'styled-jsx/css'

export default css`
  .DetailModal {
    display: block;
    position: absolute;
    right: 0;
    top: 0;
    bottom: 0;
    left: calc(30% + 15vw);
    height: 100%;
    transition: -webkit-clip-path 1s, clip-path 1s;
    transform: translateX(0);
    clip-path: circle(150% at bottom right);
    font-size: 0.8em;
    z-index: 3;
  }
  
  .content {
    margin: 7.5vh;
    mix-blend-mode: color-burn;
  }
  
  .description {
    border-width: 1px 0 1px 0;
    border-color: #444;
    border-style: solid;
    color: #333;
    font-weight: lighter;
    line-height: 1.8;
    letter-spacing: 0.1em;
  }
  
  .DetailModal.is-closed {
    transform: translateX(100%);
    clip-path: circle(0px at bottom right);
    transition: -webkit-clip-path 1s, clip-path 1s, transform 0.01s 1s;
  }

  .close-icon:hover > :global(svg) {
    opacity: 0.5;
    transition: 0.5s;
    cursor: pointer;
  }

  .close-icon > :global(svg) {
    stroke: #333;
    stroke-width: 1.5;
  }

  .close-icon {
    padding: 0;
    appearance: none;
    border-width: 0;
    font-size: 1.1em;
    margin-top: 0.5em;
    float: right;
  }

`
