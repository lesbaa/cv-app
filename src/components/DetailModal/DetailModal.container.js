import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import {
  showSkillDetailModal,
  hideSkillDetailModal,
} from '~/actionCreators/modals.js'
import DetailModal from './DetailModal.component'

const mapStateToProps = ({
  detailsModal: {
    modalIsVisible,
    modalTitle,
    modalContent,
    originX,
    originY,
  },
}, ownProps) => ({
  modalIsVisible,
  modalTitle,
  modalContent,
  originX,
  originY,
  ...ownProps,
})

const mapDispatchToProps = dispatch => bindActionCreators({
  showSkillDetailModal,
  hideSkillDetailModal,
}, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(DetailModal)

