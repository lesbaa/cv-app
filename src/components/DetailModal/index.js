import DetailModal from './DetailModal.container'

export default DetailModal
