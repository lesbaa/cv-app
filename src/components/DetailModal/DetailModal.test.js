/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'

import DetailModal from './DetailModal.component'

describe('DetailModal', () => {

  const mockCloseModal = jest.fn()

  const wrapper = shallow(
    <DetailModal
      skillId="hello"
      modalTitle="TITLE"
      modalContent="<p>content markup in here from backend</p>"
      hideSkillDetailModal={mockCloseModal}
      palette={{
        gradient: 'to bottom right, #eee6ee, #d9d9d9',
        colors: [
          '#eee6ee',
          '#d9d9d9',
        ],
      }}
    />
  )

  it('matches its snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders applies a className "is-closed" if modalIsVisible is falsey', () => {
    const { className } = wrapper.props()

    expect(className).toMatch('is-closed')
  })

  it('calls the hideSkillDetailModal action', () => {
    const closeButton = wrapper
      .find('button')

    closeButton.simulate('click')

    expect(mockCloseModal).toHaveBeenCalled()
  })

})
