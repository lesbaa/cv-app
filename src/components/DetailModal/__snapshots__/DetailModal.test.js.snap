// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailModal matches its snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailModal
    hideSkillDetailModal={[MockFunction]}
    modalContent="<p>content markup in here from backend</p>"
    modalIsVisible={false}
    modalTitle="TITLE"
    palette={
      Object {
        "colors": Array [
          "#eee6ee",
          "#d9d9d9",
        ],
        "gradient": "to bottom right, #eee6ee, #d9d9d9",
      }
    }
    skillId="hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "hello",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="jsx-779896133 content"
        >
          <Title
            title="TITLE"
          />
          <div
            className="jsx-779896133 description"
          >
            <span
              className="jsx-779896133"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>content markup in here from backend</p>",
                }
              }
            />
             
          </div>
          <button
            className="jsx-779896133 close-icon"
            onClick={[MockFunction]}
          >
            <X
              color="currentColor"
              size="24"
            />
          </button>
        </div>,
        <JSXStyle
          css=".DetailModal.jsx-779896133{display:block;position:absolute;right:0;top:0;bottom:0;left:calc(30% + 15vw);height:100%;-webkit-transition:-webkit-clip-path 1s,clip-path 1s;transition:-webkit-clip-path 1s,clip-path 1s;-webkit-transform:translateX(0);-ms-transform:translateX(0);transform:translateX(0);-webkit-clip-path:circle(150% at bottom right);clip-path:circle(150% at bottom right);font-size:0.8em;z-index:3;}.content.jsx-779896133{margin:7.5vh;mix-blend-mode:color-burn;}.description.jsx-779896133{border-width:1px 0 1px 0;border-color:#444;border-style:solid;color:#333;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}.DetailModal.is-closed.jsx-779896133{-webkit-transform:translateX(100%);-ms-transform:translateX(100%);transform:translateX(100%);-webkit-clip-path:circle(0px at bottom right);clip-path:circle(0px at bottom right);-webkit-transition:-webkit-clip-path 1s,clip-path 1s,-webkit-transform 0.01s 1s;-webkit-transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;}.close-icon.jsx-779896133:hover>svg{opacity:0.5;-webkit-transition:0.5s;transition:0.5s;cursor:pointer;}.close-icon.jsx-779896133>svg{stroke:#333;stroke-width:1.5;}.close-icon.jsx-779896133{padding:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;border-width:0;font-size:1.1em;margin-top:0.5em;float:right;}"
          styleId="779896133"
        />,
      ],
      "className": "jsx-779896133 DetailModal is-closed",
      "style": Object {
        "background": "linear-gradient(to bottom right, #eee6ee, #d9d9d9)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Title
              title="TITLE"
            />,
            <div
              className="jsx-779896133 description"
            >
              <span
                className="jsx-779896133"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>content markup in here from backend</p>",
                  }
                }
              />
               
            </div>,
            <button
              className="jsx-779896133 close-icon"
              onClick={[MockFunction]}
            >
              <X
                color="currentColor"
                size="24"
              />
            </button>,
          ],
          "className": "jsx-779896133 content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "TITLE",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="jsx-779896133"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>content markup in here from backend</p>",
                    }
                  }
                />,
                " ",
              ],
              "className": "jsx-779896133 description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "jsx-779896133",
                  "dangerouslySetInnerHTML": Object {
                    "__html": "<p>content markup in here from backend</p>",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <X
                color="currentColor"
                size="24"
              />,
              "className": "jsx-779896133 close-icon",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "currentColor",
                "size": "24",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".DetailModal.jsx-779896133{display:block;position:absolute;right:0;top:0;bottom:0;left:calc(30% + 15vw);height:100%;-webkit-transition:-webkit-clip-path 1s,clip-path 1s;transition:-webkit-clip-path 1s,clip-path 1s;-webkit-transform:translateX(0);-ms-transform:translateX(0);transform:translateX(0);-webkit-clip-path:circle(150% at bottom right);clip-path:circle(150% at bottom right);font-size:0.8em;z-index:3;}.content.jsx-779896133{margin:7.5vh;mix-blend-mode:color-burn;}.description.jsx-779896133{border-width:1px 0 1px 0;border-color:#444;border-style:solid;color:#333;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}.DetailModal.is-closed.jsx-779896133{-webkit-transform:translateX(100%);-ms-transform:translateX(100%);transform:translateX(100%);-webkit-clip-path:circle(0px at bottom right);clip-path:circle(0px at bottom right);-webkit-transition:-webkit-clip-path 1s,clip-path 1s,-webkit-transform 0.01s 1s;-webkit-transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;}.close-icon.jsx-779896133:hover>svg{opacity:0.5;-webkit-transition:0.5s;transition:0.5s;cursor:pointer;}.close-icon.jsx-779896133>svg{stroke:#333;stroke-width:1.5;}.close-icon.jsx-779896133{padding:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;border-width:0;font-size:1.1em;margin-top:0.5em;float:right;}",
          "styleId": "779896133",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "hello",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="jsx-779896133 content"
          >
            <Title
              title="TITLE"
            />
            <div
              className="jsx-779896133 description"
            >
              <span
                className="jsx-779896133"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>content markup in here from backend</p>",
                  }
                }
              />
               
            </div>
            <button
              className="jsx-779896133 close-icon"
              onClick={[MockFunction]}
            >
              <X
                color="currentColor"
                size="24"
              />
            </button>
          </div>,
          <JSXStyle
            css=".DetailModal.jsx-779896133{display:block;position:absolute;right:0;top:0;bottom:0;left:calc(30% + 15vw);height:100%;-webkit-transition:-webkit-clip-path 1s,clip-path 1s;transition:-webkit-clip-path 1s,clip-path 1s;-webkit-transform:translateX(0);-ms-transform:translateX(0);transform:translateX(0);-webkit-clip-path:circle(150% at bottom right);clip-path:circle(150% at bottom right);font-size:0.8em;z-index:3;}.content.jsx-779896133{margin:7.5vh;mix-blend-mode:color-burn;}.description.jsx-779896133{border-width:1px 0 1px 0;border-color:#444;border-style:solid;color:#333;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}.DetailModal.is-closed.jsx-779896133{-webkit-transform:translateX(100%);-ms-transform:translateX(100%);transform:translateX(100%);-webkit-clip-path:circle(0px at bottom right);clip-path:circle(0px at bottom right);-webkit-transition:-webkit-clip-path 1s,clip-path 1s,-webkit-transform 0.01s 1s;-webkit-transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;}.close-icon.jsx-779896133:hover>svg{opacity:0.5;-webkit-transition:0.5s;transition:0.5s;cursor:pointer;}.close-icon.jsx-779896133>svg{stroke:#333;stroke-width:1.5;}.close-icon.jsx-779896133{padding:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;border-width:0;font-size:1.1em;margin-top:0.5em;float:right;}"
            styleId="779896133"
          />,
        ],
        "className": "jsx-779896133 DetailModal is-closed",
        "style": Object {
          "background": "linear-gradient(to bottom right, #eee6ee, #d9d9d9)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Title
                title="TITLE"
              />,
              <div
                className="jsx-779896133 description"
              >
                <span
                  className="jsx-779896133"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>content markup in here from backend</p>",
                    }
                  }
                />
                 
              </div>,
              <button
                className="jsx-779896133 close-icon"
                onClick={[MockFunction]}
              >
                <X
                  color="currentColor"
                  size="24"
                />
              </button>,
            ],
            "className": "jsx-779896133 content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "TITLE",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="jsx-779896133"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<p>content markup in here from backend</p>",
                      }
                    }
                  />,
                  " ",
                ],
                "className": "jsx-779896133 description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "jsx-779896133",
                    "dangerouslySetInnerHTML": Object {
                      "__html": "<p>content markup in here from backend</p>",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <X
                  color="currentColor"
                  size="24"
                />,
                "className": "jsx-779896133 close-icon",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "currentColor",
                  "size": "24",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".DetailModal.jsx-779896133{display:block;position:absolute;right:0;top:0;bottom:0;left:calc(30% + 15vw);height:100%;-webkit-transition:-webkit-clip-path 1s,clip-path 1s;transition:-webkit-clip-path 1s,clip-path 1s;-webkit-transform:translateX(0);-ms-transform:translateX(0);transform:translateX(0);-webkit-clip-path:circle(150% at bottom right);clip-path:circle(150% at bottom right);font-size:0.8em;z-index:3;}.content.jsx-779896133{margin:7.5vh;mix-blend-mode:color-burn;}.description.jsx-779896133{border-width:1px 0 1px 0;border-color:#444;border-style:solid;color:#333;font-weight:lighter;line-height:1.8;-webkit-letter-spacing:0.1em;-moz-letter-spacing:0.1em;-ms-letter-spacing:0.1em;letter-spacing:0.1em;}.DetailModal.is-closed.jsx-779896133{-webkit-transform:translateX(100%);-ms-transform:translateX(100%);transform:translateX(100%);-webkit-clip-path:circle(0px at bottom right);clip-path:circle(0px at bottom right);-webkit-transition:-webkit-clip-path 1s,clip-path 1s,-webkit-transform 0.01s 1s;-webkit-transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;transition:-webkit-clip-path 1s,clip-path 1s,transform 0.01s 1s;}.close-icon.jsx-779896133:hover>svg{opacity:0.5;-webkit-transition:0.5s;transition:0.5s;cursor:pointer;}.close-icon.jsx-779896133>svg{stroke:#333;stroke-width:1.5;}.close-icon.jsx-779896133{padding:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;border-width:0;font-size:1.1em;margin-top:0.5em;float:right;}",
            "styleId": "779896133",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
